tuist/tuist
# Add `Workspace.GenerationOptions` to support `WorkspaceSettings.xcsettings`.

Resolves https://github.com/tuist/tuist/issues/3487

### Short description 📝

Draft PR for early feedback. Things left to resolve:
- [x] Encode/Decode issue from `ProjectDescription.Workspace` to `TuistGraph.Workspace`.
- [x] Need the [XcodeProj PR](https://github.com/tuist/XcodeProj/pull/658) merged and released (currently pointing to my fork for testing).

To provide a mechanism to write the `WorkspaceSettings.xcsettings` file in order to silence the Xcode prompt about how it should handle schemes.

### How to test the changes locally 🧐

Update your local `Workspace.swift` manifest to pass the new `[GenerationOptions]`, passing the `automaticSchemaGenerations` value. To see non-default behavior, pass either `.enabled` or `.disabled` associated values. Then generate your project. You should see a new file: `[project].xcworkspace/WorkspaceSettings.xcsettings`

### Checklist ✅
